### PR TITLE
STUD-286: clean up teams screen

### PR DESCRIPTION
**Pull request by yasfahm**
- Cleaned up teams screen
- Combined all different components on the same screen with their respective functionalities (tabs, create team button, filter and search buttons)
- Placed hard coded ParticipantListItem on participants tab and TeamListItem  just for **visualization**
- Placed create team button on teams tab

- Changed styling of join team button to match the prototype

Screen Shot 2022-03-24 at 3 04 16 PM
Screen Shot 2022-03-24 at 3 04 29 PM
Screen Shot 2022-03-24 at 3 04 47 PM
Screen Shot 2022-03-24 at 3 04 55 PM

IMG_380892C923B2-1

### CODE REVIEW

File: src/containers/ListContainer.js [hunk 1]
```diff
@@ -10,141 +10,215 @@ import { Pressable } from 'react-native';
...
+        this.deleteLabel = this.deleteLabel.bind(this);
+        this.searchBarVisibility = this.searchBarVisibility.bind(this);
     }
 
-    onToggle() {
-        this.setState({isToggled: !this.state.isToggled});
-    }
+    deleteLabel = () => {
+        this.setState({ showLabel: false });
+        console.log("Deleted");
```
Comment (nuzurie) [inline src/containers/ListContainer.js]: remove pls

File: src/components/SearchBar.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  
+    this.state = {
+        searchBarVisibility: false,
+        participantName: '',
+    };
+    this.triggerSearchBar = this.triggerSearchBar.bind(this);
+  }
+  // Write functions here
+  triggerSearchBar() {
+    this.setState({ searchBarVisibility: !this.state.searchBarVisibility });
```
Comment (nuzurie) [inline src/components/SearchBar.js:24]: use arrow function syntax:
```
this.setState(prevState => { searchBarVisibility: !prevState.searchBarVisibility });
```

File: src/components/SearchBar.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { Title, Subtitle, TextBody } from '../containers/TextContainer.js';
+import MainContainer from '../containers/MainContainer.js';
+import JoinTeam from '../components/modals/JoinTeam.js';
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+  
+    this.state = {
+        searchBarVisibility: false,
```
Comment (nuzurie) [inline src/components/SearchBar.js:17]: small suggestion: make this `searchBarVisible`

File: src/components/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  }
+  render() {
+    return (
+        <Search>
+            <Pressable onPress={this.triggerSearchBar}>
+                <SearchIcon source={MagnifyingIcon}></SearchIcon>
+            </Pressable>
+            { this.state.searchBarVisibility && 
+                /** Show search bar on click */
+                <SearchField placeholder="Search" onChangeText={(text) => this.participantName = text} value={this.state.participantName}/>
```
Comment (nuzurie) [inline src/components/SearchBar.js:34]: Does this work? 
`onChangeText={(text) => this.participantName = text}`
If you're trying to change the state, why not use setState? Or is this some other variable I'm missing

File: src/containers/ListContainer.js [hunk 5]
```diff
@@ -10,141 +10,215 @@ import { Pressable } from 'react-native';
...
+        this.className.clear();
+      }
+
+      handleSearchSubmit = () => {
+        // TODO
+        this.participantName.clear();
+
+      }
+      filterBarVisibility = () => {
+        this.setState({ selectedButton: this.state.selectedButton = 1})
```
Comment (nuzurie) [inline src/containers/ListContainer.js]: what does this do? `this.setState({ selectedButton: this.state.selectedButton = 1})`
How is it different from `this.setState({ selectedButton: 1})`

File: src/containers/ListContainer.js [hunk 6]
```diff
@@ -10,141 +10,215 @@ import { Pressable } from 'react-native';
...
+        this.setState({ classEntered: !this.state.classEntered });
```
Comment (nuzurie) [inline src/containers/ListContainer.js:54]: use arrow function pls

File: src/containers/ListContainer.js [hunk 7]
```diff
@@ -10,141 +10,215 @@ import { Pressable } from 'react-native';
...
+      }
+    
+      searchBarVisibility = () => {
```
Comment (nuzurie) [inline src/containers/ListContainer.js]: same as filter

File: src/containers/ListContainer.js [hunk 8]
```diff
@@ -10,141 +10,215 @@ import { Pressable } from 'react-native';
...
-            <Icon source={Selection} />
-            </Header>
+                <Header>
+                  <SearchButton onPress={this.searchBarVisibility}>
+                    <SearchIcon source={MagnifyingIcon} />
+                  </SearchButton>
+                  <FilterButton onPress={this.filterBarVisibility}>
+                      <IconFilter source={Sort} />
+                  </FilterButton>
+                  { this.state.selectedButton === 1 ?
```
Comment (nuzurie) [inline src/containers/ListContainer.js]: Why don't we just make it a boolean?

File: src/screens/TeamFormation/Tabs/Participants.js [hunk 9]
```diff
@@ -16,20 +12,25 @@ class Participants extends Component {
...
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-          <MainContainer><Subtitle>John</Subtitle></MainContainer><Text>{'\n'}</Text>
-
+           {/* placing these for visualization */}
+          <ParticipantListItem
```
Comment (nuzurie) [inline src/screens/TeamFormation/Tabs/Participants.js:16]: great job!

File: src/screens/TeamFormation/Tabs/Teams.js [hunk 10]
```diff
@@ -1,65 +1,48 @@
...
-          <Text>{'\n'}</Text>
-          <Text>{'\n'}</Text>
-          <Text>{'\n'}</Text>
-          <Text>{'\n'}</Text>
-
-          </ListContainer>
-                
-           
+             {/* placing these for visualization */}
+            <TeamListItem
```
Comment (nuzurie) [inline src/screens/TeamFormation/Tabs/Teams.js:20]: nice!

**Review by nuzurie — APPROVED** [review]

File: src/containers/ListContainer.js (hunk 8 above)
Comment (yasfahm) [reply src/containers/ListContainer.js]: great idea!

File: src/containers/ListContainer.js (hunk 5 above)
Comment (yasfahm) [reply src/containers/ListContainer.js]: changed it to boolean

**Pull request merged**